Feature/drop support of php 7, use php 8 language level features

Chris53897 opened the pull request:
This is target for a new major version 2.x There are BC breaks.

- drop support of php 7
- use language level features of php 8
- add strong types
- remove some phpdoc comments

There are some paramters i am not sure if nullable or not.

Can the deprecated functions be removed? I do not yet how much work that will be.

WDYT?

bojanz commented:
@Chris53897 It's finally time to open 2.x. I apologize for this taking so long. Could you do one last push to remove the travis changes (we've moved to GitHub actions)? I can then proceed with the merge. Thank you!

Chris53897 commented:
I removed `.travis` and improved `phpunit.xml` (the default-values are removed).

But tests failed beacuse of `string` vs `?string` in `AddressInterface` vs `Address.php`.
I changed the interface. Please have a look if this is correct.

bojanz commented:
Thank you!

?string is definitely the safer default so happy to proceed with that for now.
